lib/python/treadmill_aws/hostmanager.py

```diff
@@ -47,19 +48,34 @@ def create_host(ec2_conn, ipa_client, image_id, count, domain,
         host_ctx['otp'] = ipa_host['result']['result']['randompassword']
         user_data = render_manifest(host_ctx)
 
-        ec2client.create_instance(
-            ec2_conn,
-            user_data=user_data,
-            image_id=image_id,
-            instance_type=instance_type,
-            key=key,
-            tags=_instance_tags(host_ctx['hostname'], role),
-            secgroup_ids=secgroup_ids,
-            subnet_id=subnet_id,
-            instance_profile=instance_profile,
-            disk=disk
-        )
-        hosts.append(host_ctx['hostname'])
+        try:
+            ec2client.create_instance(
+                ec2_conn,
+                user_data=user_data,
+                image_id=image_id,
+                instance_type=instance_type,
+                key=key,
+                tags=_instance_tags(host_ctx['hostname'], role),
+                secgroup_ids=secgroup_ids,
+                subnet_id=subnet_id,
+                instance_profile=instance_profile,
+                disk=disk
+            )
+            hosts.append(host_ctx['hostname'])
+
+        except ClientError as error:
+            error_code = error.response['Error']['Code']
+
+            if error_code == 'InsufficientFreeAddressesInSubnet':
+                try:
+                    ipa_client.unenroll_host(hostname=host_ctx['hostname'])
+                except (KeyError, ipaclient.NotFoundError):
+                    pass
+
+                break
+
+            else:
+                raise error
```

Comment:
you should change the docstring for the function, describing new behavior. should also add unittest for that.